frontend/src/app/annotate/annotate.component.html

```diff
@@ -1,22 +1,93 @@
+        } @else if (adding) {
+        <section class="d-flex flex-column gap-2">
+            <p class="text-muted" i18n>
+                You are currently adding a new problem.
+            </p>
+            <p class="text-muted" i18n>
+                Hit 'Start Parse' to send your problem data to the parser and
+                inspect the results.
+            </p>
+            <p class="text-muted" i18n>
+                If you are happy with the changes, save the problem by clicking
+                the 'Save' button.
+            </p>
+            <p class="text-muted" i18n>
+                This will add the problem to the database.
+            </p>
+        </section>
+        } @else if (editing) {
+        <section>
+            <p class="text-muted" i18n>
+                You are currently editing an existing problem.
+            </p>
+            <p class="text-muted" i18n>
+                Hit 'Start Parse' to send your problem data to the parser and
+                inspect the results.
+            </p>
+            <p class="text-muted" i18n>
+                If you are happy with the changes, save the changes by clicking
+                the 'Save problem' button.
+            </p>
+            @if (isUserProblem$ | async) {
+            <p class="text-muted" i18n>
+                This will update this problem in the database.
+            </p>
+            } @else {
+            <p class="text-muted" i18n>
+                This will create a new problem in the database, leaving the
+                original problem unchanged.
+            </p>
+            }
+        </section>
+        } @else {
+        <section class="d-flex justify-content-center align-items-center">
+            <div class="spinner-border spinner-border-lg">
+                <span class="visually-hidden" i18n>Loading...</span>
+            </div>
+        </section>
+        }
+        <div class="d-flex flex-column gap-2 align-items-stretch">
+            @if (browsing) {
             <button
-                type="submit"
-                class="btn btn-success btn-lg d-flex align-items-center"
+                type="button"
+                class="btn btn-secondary d-flex align-items-center justify-content-center"
+                (click)="addProblem()"
             >
                 <fa-icon
-                    [icon]="faTree"
-                    class="fa-lg"
+                    [icon]="faPlus"
                     aria-hidden="true"
                     focusable="false"
                 ></fa-icon>
-                <span class="ms-4 fs-3" i18n>Start parse</span>
+                <span class="ms-3" i18n>Add new problem</span>
             </button>
+            } @else {
+            <button
+                type="button"
+                class="btn btn-secondary d-flex align-items-center justify-content-center"
+                (click)="firstProblemId && goToProblem(firstProblemId)"
+            >
```

Comment:
I'm a bit rusty on Angular, so I might be going against the conventions and/or asking for the impossible, but my acquired preferences would tell me to rather do something like this instead,

```suggestion
            @if (firstProblemId) {
            <a
                class="btn btn-secondary d-flex align-items-center justify-content-center"
                href="/annotate/{{firstProblemId}}"
            >
```

so I wouldn't need to inject the router in the component and basically do the router's work by making it navigate manually. Is that conceivable?

Reply:
BTW this is an incomplete change suggestion because I was too lazy to go all the way to the closing `</a>}`.

Reply:
No, you're right! All this button does is navigating, so I agree it is best replaced by an anchor tag.

Reply:
Same goes for the one above it. I'll run a global check to see if other buttons in this PR can be replaced with `<a>`s for semantic reasons.